pkg/kube/inject/webhook.go

```diff
@@ -608,10 +610,45 @@ func applyPrometheusMerge(pod *corev1.Pod, mesh *meshconfig.MeshConfig) error {
 		pod.Annotations["prometheus.io/port"] = targetPort
 		pod.Annotations["prometheus.io/path"] = "/stats/prometheus"
 		pod.Annotations["prometheus.io/scrape"] = "true"
```

Comment:
What if a user sets `prometheus/io/path: foo`. Then we add `prometheus.io/path: /stats/prometheus`. Then won't we end up with *both*, at which point prometheus may pick a random one?

We may need to clear out all variants and then set ours?

Reply:
```golang
        for k, v := range pod.Annotations {
		ln := strutil.SanitizeLabelName(k)
		ls[model.LabelName(podAnnotationPrefix+ln)] = lv(v)
		ls[model.LabelName(podAnnotationPresentPrefix+ln)] = presentValue
	}
```
I think prometheus will take a random one due to range from a map 

I prefer to merge istio metric to the path which's already settled in annotations, wdyk @howardjohn

Reply:
```
We may need to clear out all variants and then set ours?
```
seems reasonable, Is there something told end users about this auto-merge behavior?

Reply:
We do say https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging they will be overriden so it seems ok to override? WDYT

Reply:
updated